src/commands/scaling/addScaleRule/AddNewScaleRule.ts

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ScaleRule, Template } from "@azure/arm-appcontainers";
+import { AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { Progress, ProgressLocation, window } from "vscode";
+import { ScaleRuleTypes } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { ContainerAppTreeItem } from "../../../tree/ContainerAppTreeItem";
+import { RevisionTreeItem } from "../../../tree/RevisionTreeItem";
+import { ScaleRuleGroupTreeItem } from '../../../tree/ScaleRuleGroupTreeItem';
+import { localize } from "../../../utils/localize";
+import { updateContainerApp } from "../../updateContainerApp";
+import { IAddScaleRuleWizardContext } from "./IAddScaleRuleWizardContext";
+
+export class AddNewScaleRule extends AzureWizardExecuteStep<IAddScaleRuleWizardContext> {
```

Comment:
Some of this logic could be moved to `ScaleRuleGroupTreeItem.createChildImpl`. If you look at `ManagedEnvironmentTreeItem`, you'll see what I mean, but the benefit of doing this is that it'll automatically handle adding the child to the parent in the UI. You will also see the "Creating..." node under the parent as it's creating.

If you need more context/have questions, let's talk about it offline.